modules/sw360/sw360-client/src/main/java/org/eclipse/sw360/antenna/sw360/client/config/SW360ClientConfig.java

```diff
@@ -115,14 +121,15 @@ private SW360ClientConfig(URI baseURI, String authURL, String user, String passw
      */
     public static SW360ClientConfig createConfig(String restURL, String authURL, String user,
                                                  String password, String clientId, String clientPassword,
-                                                 HttpClient httpClient, ObjectMapper mapper) {
+                                                 String token, HttpClient httpClient, ObjectMapper mapper) {
         return new SW360ClientConfig(
                 URI.create(stripTrailingSeparator(Validate.notEmpty(restURL, "Undefined REST URL"))),
                 stripTrailingSeparator(Validate.notEmpty(authURL, "Undefined authentication URL")),
                 Validate.notEmpty(user, "Undefined user"),
                 Validate.notEmpty(password, "Undefined password"),
                 Validate.notEmpty(clientId, "Undefined client ID"),
                 Validate.notEmpty(clientPassword, "Undefined client password"),
+                Validate.notEmpty(token, "Undefined token"),
                 Validate.notNull(httpClient),
```

Comment:
We now support multiple ways of authentication that make use of different properties; therefore, we should force that all these properties are defined. The validation logic should be adapted to check that either user name and password are defined or the access token (and probably not both).

Reply:
change the validation based on token